core/core-funcs.el

```diff
@@ -253,5 +253,11 @@ result, incrementing passed-tests and total-tests."
      (concat "Hidden Mode Line Mode enabled.  "
              "Use M-x hidden-mode-line-mode to make the mode-line appear."))))
 
+(defun spacemacs/recompile-elpa ()
+  "Recompile packages in elpa directory. Useful if you switch
+Emacs versions."
+  (interactive)
+  (byte-recompile-directory package-user-dir nil t))
```

Comment:
Optional arguments are at the end, and usually they are used to change expected behaviours or default values. So when I read in the PR:

    (byte-recompile-directory package-user-dir nil t)

I know you are calling the recompilation, but with a non-default behaviour. I had to open spacemacs to look at the function documentation to know what was the purpose of the last `t`. If you were instead using `'force` (or `:force`) I would have know directly the goal is to force recompilation. For the `nil` argument, it means that an optional argument is not "activated", so you're not changing a default behaviour, so I don't really care what it is.

When you are reading through the source code, if this trick is used every time, it is directly obvious what a code is doing. With `t` everywhere you'll have to open the documentation of all functions.

The same is also valid for other examples. For instance assume you don't know what's the syntax of `add-hook`. If you read this in spacemacs source:

    (add-hook 'some-mode-hook 'some-function t)

you will have to check what's the purpose of the `t` argument. On the other hand if you read:

    (add-hook 'some-mode-hook 'some-function 'append)

you know that the goal is to append the hook instead of pushing it at beginning.